### PR TITLE
Sometimes toast about old join community notification starts appearin…

### LINKED ISSUE #17955: Sometimes toast about old join community notification starts appearing on every login
There are no reproduction steps yet. Hope attached logs might help. 

For now I have faced the issue only on Android but I don't think it is device specific. It is hard to verify as it happens randomly.

On the video below (recorded on release build) you can see the toast appearing on every relogin. 

[Status-debug-logs (83).zip](https://github.com/status-im/status-mobile/files/13426193/Status-debug-logs.83.zip)

https://github.com/status-im/status-mobile/assets/97245802/bd716afe-e0fe-4bab-80d9-addf26bf1587

### Additional Information

- Status version: release 1.26, nightly
- Operating System: Android

### PR DESCRIPTION
**Pull request by ibrkhalil**
### Summary

fixes #17955

Returns a notification on the MessengerResponse only when joining a community.

status: ready

### CODE REVIEW

File: src/status_im/contexts/communities/events.cljs [hunk 1]
```diff
@@ -88,6 +83,13 @@
   (let [my-public-key (get-in db [:profile/profile :public-key])]
     {:db (reduce (fn [db {:keys [public-key] :as request}]
                    (let [my-request? (= my-public-key public-key)]
+                     (log/info "Handling community request to join "
```
Comment (cammellos) [inline src/status_im/contexts/communities/events.cljs]: not needed

File: src/status_im/contexts/profile/push_notifications/local/events.cljs [hunk 2]
```diff
@@ -21,6 +24,14 @@
      (or (= app-state "background")
          (not (foreground-chat? cofx chat-id))))))
 
+(defn show-community-joined-toast
+  [{:keys [bodyType title]}]
+  (when (= bodyType "communityJoined")
```
Comment (cammellos) [inline src/status_im/contexts/profile/push_notifications/local/events.cljs]: please const this in the `constants` file

File: src/status_im/contexts/profile/push_notifications/local/events.cljs [hunk 3]
```diff
@@ -32,7 +43,9 @@
    bodyType))
 
 (rf/defn process
-  [cofx event]
-  (if platform/ios?
-    {:effects/push-notifications-local-present-ios (create-notification nil event)}
-    {:effects/push-notifications-local-present-android (create-notification cofx event)}))
+  [cofx {:keys [bodyType] :as event}]
+  (if (push-notification-types bodyType)
```
Comment (cammellos) [inline src/status_im/contexts/profile/push_notifications/local/events.cljs:48]: make sure that you captured all the types in the list please (looks like, but double check).

**Review by ilmotta — APPROVED** [review]
Ignoring @cammellos' comments, the rest LGTM

File: src/status_im/contexts/profile/push_notifications/local/events.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/status_im/contexts/profile/push_notifications/local/events.cljs]: Done

File: src/status_im/contexts/communities/events.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/status_im/contexts/communities/events.cljs]: Removed

File: src/status_im/contexts/profile/push_notifications/local/events.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/status_im/contexts/profile/push_notifications/local/events.cljs:48]: Checked, Thanks

**Review by cammellos — APPROVED** [review]

**Comment by pavloburykh** [comment]
@ibrkhalil thank you for the PR. Could you please rebase go and mobile branches so we proceed with testing?

**Comment by pavloburykh** [comment]
@ibrkhalil thank you for the PR. 

The situation with toasts seems to be better but unfortunately I still randomly receive old toasts and AC notifications. Please, see description in ISSUE 1 below

### ISSUE 1 Old contact request toast + community AC notification appears every time user leaves specific community

I am not sure that it can be replicated easily following some steps, as in my case it is reproducible on Community A but is not reproducible with community B. 

**Preconditions:** 

- 'Sepolia community 4' is token gated community with requirement of manual approval of community requests.
- User A is eligible to join 'Sepolia community 4' as Admin/Member. 

**On the video below you see the following:** 

1. User A sends join request to Community 'Sepolia community 4' (this is token
2. Community admin accepts the request and User A becomes community member
3. User A leaves 'Sepolia community 4'

**Actual result:** 

1. User A receives AC notification about joining  'Sepolia community 4' despite he has left it
2. User A receives a toast about User B has accepted his contact request despite User B is already a contact and no additional requests have been sent/accepted

This behaviour repeats every time User A leaves  'Sepolia community 4'. 

@ibrkhalil ping me up and I will provide a seed phrase of User A. Maybe you will be able to replicate the issue in restored user. 

[Status-debug-logs - 2024-03-19T140256.263.zip](https://github.com/status-im/status-mobile/files/14650232/Status-debug-logs.-.2024-03-19T140256.263.zip)

https://github.com/status-im/status-mobile/assets/97245802/65422241-da8f-4228-8d20-cd1b438eba90

**Comment by pavloburykh** [comment]
@ibrkhalil please ping me up when PR is ready for re-test. Thank you.

**Comment by ibrkhalil** [comment]
> @ibrkhalil please ping me up when PR is ready for re-test. Thank you.

I had trouble with reproducing consistently to debugging, I'll try again later today and ping you.
Thanks Pavlo!

**Comment by ibrkhalil** [comment]
> @ibrkhalil please ping me up when PR is ready for re-test. Thank you.

Hey Pavlo, I fixed the issue with the AC notifications not being cleared after leaving the community and couldn't reproduce the other issue.
Can you please confirm if it still exists?
And do you think it's better to be another issue, Since it's related to contact requests and not communities?

**Comment by ibrkhalil** [comment]
> > @ibrkhalil please ping me up when PR is ready for re-test. Thank you.
> 
> Hey Pavlo, I fixed the issue with the AC notifications not being cleared after leaving the community and couldn't reproduce the other issue. Can you please confirm if it still exists? And do you think it's better to be another issue, Since it's related to contact requests and not communities?

Also, I realized that the 'Manage communities on testnet' option on status-desktop settings get disabled after some time, And notifications can be weird if I didn't enable it it back.

**Comment by pavloburykh** [comment]
@ibrkhalil thank you for your fixes. Unfortunately I am still facing many weird issues related to community system notifications. 

Here are some example from this PR: 

1. I am receiving 'You joined community' notification despite I have just left community
2. After joining Community X for some reason I am receiving notification about acceptance of join request in Community Y despite I am not even a member of Community Y anymore.
3. After joining Community X I am receiving bunch of notifications with mentions/replies which I have previously received and read in this community before I have left and re-joined it. 
4. Very often user does not receive notification about joining the community.
5. User almost never receives notification about kicking from community. 

https://github.com/status-im/status-mobile/assets/97245802/c90f4442-2d4c-4efe-ad97-55d2980eff44

Overall, cases related to system community notifications are in complete mess. I don't think we have time to investigate and fix them in this release, so I propose not to include this PR into 2.28 and try to fix all those issues in 2.29. 

WDYT @cammellos?

**Comment by cammellos** [comment]
@pavloburykh yes, I think that's ok, 2.28.1 or 2.29, we won't have time to fix it properly

**Comment by ibrkhalil** [comment]
Closing for now

**Pull request closed without merging**